### PR TITLE
Allow setting of prompt in the shell session.

### PR DESCRIPTION
**Pull request by andrm**
Subject says it all. There are some casts in it, please provide guidance
on how to avoid it.

### CODE REVIEW

**Comment by vietj** [comment]
I think we could instead use a function like on `Shell` to be more dynamic

```
void setPrompt(Function<Session, String> prompt);
```

that would be evaluated when the shell prompt is required.

**Comment by andrm** [comment]
Cool. Will redo it.

File: src/main/java/io/vertx/ext/shell/session/Session.java [hunk 1]
```diff
@@ -79,4 +81,14 @@ static Session create() {
...
 
+
+  /**
+   * Set a new prompt in this session.
+   *
+   * @param prompt  the new prompt will be calculated when it's needed.
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  Session setPrompt(Function<Session, String> prompt);
```
Comment (vietj) [inline src/main/java/io/vertx/ext/shell/session/Session.java]: please move to the `Shell` interface

Comment (andrm) [reply src/main/java/io/vertx/ext/shell/session/Session.java]: Done.

**Comment by vietj** [comment]
@andrm have you signed the eclipse contributor ?

**Comment by andrm** [comment]
Not yet. How do I do this?

**Comment by andrm** [comment]
Will do.

**Comment by andrm** [comment]
Done.

**Comment by vietj** [comment]
great contrib, thanks @andrm

**Pull request merged**